app/util/analytics.py

```diff
@@ -114,13 +134,27 @@ def get_events(since_timestamp=None, since_id=None):
     since_timestamp = float(since_timestamp) if since_timestamp else 0.0
     since_id = int(since_id) if since_id else 0
 
+    filtered_events_from_cache = get_events_from_cache(since_timestamp, since_id)
+    if len(filtered_events_from_cache) > 0:
+        return filtered_events_from_cache
```

Comment:
What if the user has requested more events than are in the cache? I know it's probably not common but do you think we should support loading more than `LOG_CACHE_MAX_SIZE` events?

I think this is still better than what we have now. :smile: